tests/test_views.py

```diff
@@ -503,6 +503,97 @@ def test_bulk_create(self, api_rf):
+                is_active=True,
+            )
+        ]
+        self.user.enrollment_records.set(self.enrollment_records)
+
+    def _make_request(self, api_rf, payload=None):
+        request = api_rf.patch(
+            f"/memberships/{self.cohort.uuid}/{self.membership.id}/",
+            payload,
+            format="json",
+        )
+        force_authenticate(request, self.manager.user)
+        return self.member_update_view(
+            request, cohort_uuid=self.cohort.uuid, pk=self.membership.id
+        )
+
+    def _setup(self, with_enrollments=False):
+        self.user = factories.UserFactory()
+        self.manager = factories.PartnerManagementMembershipFactory()
+
+        self.partner = self.manager.partner
+        self.cohort = factories.PartnerCohortFactory(partner=self.partner)
+        self.membership = factories.CohortMembershipFactory(
+            cohort=self.cohort, user=self.user
+        )
+
+        if with_enrollments:
+            self._setup_enrollments()
+
+        self.member_update_view = views.CohortMembershipUpdateView.as_view()
+
+    def test_manager_can_update_membership(self, api_rf):
+        """Managers can update membership for cohorts they manage."""
+        self._setup()
+
+        response = self._make_request(api_rf, payload={"active": False})
+        assert response.status_code == 200
+        assert (
+            self.cohort.memberships.first().status
+            == enums.CohortMembershipStatus.DEACTIVATED.value
+        )
+
+    def test_course_enrollments_deactivated_on_status_change(self, api_rf):
+        """
+        Confirms enrollment is marked inactive when a user is deactivated.
+        """
+        self._setup(with_enrollments=True)
+
+        response = self._make_request(api_rf, payload={"active": False})
+        assert (
+            self.cohort.memberships.first().status
+            == enums.CohortMembershipStatus.DEACTIVATED.value
+        )
+        assert response.status_code == 200
+        assert self.user.enrollment_records.first().is_active is False
+
+    def test_course_enrollments_if_multiple_cohorts(self, api_rf):
+        """
+        Confirms enrollments are not affected if a user is a member of multiple
+        cohorts with the same cohort offering.
+        """
+        self._setup(with_enrollments=True)
+
+        # Create new cohort with existing offering in another cohort
+        other_cohort = factories.PartnerCohortFactory(partner=self.manager.partner)
+        factories.CohortMembershipFactory(cohort=other_cohort, user=self.user)
+        factories.CohortOfferingFactory(cohort=other_cohort, offering=self.offering)
+
+        response = self._make_request(api_rf, payload={"active": False})
+        assert (
+            self.cohort.memberships.first().status
+            == enums.CohortMembershipStatus.DEACTIVATED.value
+        )
+        assert response.status_code == 200
+        assert self.user.enrollment_records.first().is_active is True
```

Comment:
Curious - why is this different than `this.enrollment_records`? I was initially testing that the original object was updated but that failed.